### PR TITLE
Refactor Code Generation

### PR DESCRIPTION
**Pull request by mohsha-msft**
- [ ] The purpose of this PR is explained in this or a referenced issue.
- [ ] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [ ] Tests are included and/or updated for code changes.
- [ ] Updates to [CHANGELOG.md][] are included.
- [ ] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go
[CHANGELOG.md]: https://github.com/Azure/azure-sdk-for-go/blob/main/CHANGELOG.md

### CODE REVIEW

**Comment by mohsha-msft** [comment]
Replacing https://github.com/Azure/azure-sdk-for-go/pull/17210

**Comment by mohsha-msft** [comment]
Hey @jhendrixMSFT @seankane-msft @benbp 

Can you please review this PR. I need one review from code owners to merge it.

**Comment by mohsha-msft** [comment]
Added PR: https://github.com/Azure/azure-sdk-for-go/pull/17276 to handle UploadStreamBug

**Review by seankane-msft — APPROVED** [review]

**Pull request closed without merging**
